### ISSUE TITLE
[Spree Upgrade][BackOffice] Fix features/admin/bulk_order_management_spec - issue with variant on_hand levels (2 broken tests)

### ISSUE THREAD

**Issue by luisramos0** [openfoodfoundation/openfoodnetwork#3090]
Fix spec/features/admin/bulk_order_management_spec (2 broken tests)

There are problems related to variant on_hand levels in  spec/features/admin/bulk_order_management_spec:
```
  1) 
  As an Administrator
  I want to be able to manage orders in bulk
 altering line item properties submitting data to the server when acceptable data is sent to the server displays an update button which submits pending changes
     Failure/Error: expect(page).to have_selector "#save-bar", text: "All changes saved"
       expected to find visible css "#save-bar" with text "All changes saved" but there were no matches. Also found "Fields with red borders contain errors.", which matched the selector but not all filters.
     # ./spec/features/admin/bulk_order_management_spec.rb:157:in `block (5 levels) in <top (required)>'

  2) 
  As an Administrator
  I want to be able to manage orders in bulk
 altering line item properties submitting data to the server when unacceptable data is sent to the server displays an update button which submits pending changes
     Failure/Error: expect(page).to have_content "exceeds available stock. Please ensure line items have a valid quantity."
       expected to find text "exceeds available stock. Please ensure line items have a valid quantity." in "Logged In As : jamil_brakus@vonruedenlegros.ca Account Logout Store DASHBOARD ORDERS PRODUCTS REPORTS CONFIGURATION USERS ORDER CYCLES GROUPS ENTERPRISES CUSTOMERS ORDERS BULK ORDER MANAGEMENT SUBSCRIPTIONS Bulk Order Management What's this? Fields with red borders contain errors. START DATE END DATE PRODUCER All SHOP All ORDER CYCLE All ACTIONS COLUMNS NAME ORDER DATE PRODUCER PRODUCT: UNIT QUANTITY MAX ASK? John Doe November 19, 2018 Enterprise 54 Product #28 - 4810: 1g, S \"Product #28 - 4810 (1g, S)\" is out of stock. 5"
     # ./spec/features/admin/bulk_order_management_spec.rb:171:in `block (5 levels) in <top (required)>'
```

**Comment by luisramos0** [comment]
These specs detect 2 bugs in spree 2's AvailabilityValidator, one fixed in v2.1:
https://github.com/spree/spree/commit/131c4e23fffa4e704e1e44dc23d0af2b27df21cf#diff-2b2c15ea125101a67e11f549cc7ebfa6
and another one fixed in 2.4.0:
https://github.com/spree/spree/commit/a9d3ad91189ee11a6b05641db4c537e3a52ff773#diff-2b2c15ea125101a67e11f549cc7ebfa6

These OFN specs need both fixes.

The bugs are pretty obvious. Checking stock availability of line items of completed orders (normal quantity adjustment in the backoffice) must look at the order's inventory_units.

We need to fix this AvailabilityValidator.
On top of this we will need to adapt this same AvailabilityValidator to work with VariantOverrides...

**Comment by sauloperez** [comment]
I would cherrypick these patches to our fork's branch as we did for couple others and mention it in the fork's README to make it even more obvious in the future (that is what I did when we started off of 2.0.4)

**Comment by luisramos0** [comment]
It's complicated because the second fix (the one we need from 2.4) depends on this DB change  from v2.2: https://github.com/spree/spree/commit/dcbd5deae87f046f147b394730b8ee277e0e50df#diff-d4464f8dddd86dd147739f1fcf6d589b

Instead of going line_item.inventory_units (created in the change above), because we only have one shipment per order, we could do line_item.order.shipment.inventory_units...
